### PR TITLE
Fix permissions tests running as root (Fix #3192)

### LINKED ISSUE #3192: PermissionsTests.test_path_drop on Darwin as root is breaking
```
[ RUN      ] PermissionsTests.test_path_drop
/Users/osquery/jenkins/workspace/osqueryPullRequestOSX/TargetSystem/vm-osx12/osquery/core/tests/posix/permissions_tests.cpp:95: Failure
Value of: dropper->dropTo(getuid(), getgid())
  Actual: true
Expected: false
```

### PR DESCRIPTION
**Pull request by obelisk**
On OSX when you run as root, it doesn't care about file permissions, it will always pass. This modifies the tests to take that into account.

I also added a new test for dropping permissions and checking that it cannot access files created with high permissions.

### CODE REVIEW

File: osquery/core/tests/posix/permissions_tests.cpp [hunk 1]
```diff
@@ -91,13 +91,13 @@ TEST_F(PermissionsTests, test_path_drop) {
     EXPECT_TRUE(dropper->dropped_);
     EXPECT_EQ(dropper->to_user_, nobody->pw_uid);
 
+    // Dropping "up" to root should fail.
     // Even though this is possible and may make sense, it is confusing!
-    EXPECT_FALSE(dropper->dropTo(getuid(), getgid()));
+    EXPECT_FALSE(dropper->dropTo(0, 0));
```
Comment (poppyseedplehzr) [inline osquery/core/tests/posix/permissions_tests.cpp]: Any reason to not have both?

**Comment by poppyseedplehzr** [comment]
Would you mind being more descriptive with your title? Perhaps "Addressing small permissions fix in testing"?

**Comment by jedi22** [comment]
This was just to make sure it passed all the tests, I will rename it before merging :)

File: osquery/core/tests/posix/permissions_tests.cpp (hunk 1 above)
Comment (jedi22) [inline osquery/core/tests/posix/permissions_tests.cpp]: Because only the change works on all systems, the first doesn't work as root on OS X.

**Comment by osqueryer** [comment]
:no_good: The commit c39fe03c8ac335d0c365050f3c25251ef6b3897a (Job results: [1748](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1748/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 3823786c0758c77f2d49a3a5fb740242d69c0065 (Job results: [1826](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1826/)) failed one or more tests (**Windows**).

**Comment by jedi22** [comment]
Fixes #3192

**Comment by osqueryer** [comment]
:-1: The commit 3823786c0758c77f2d49a3a5fb740242d69c0065 (Job results: [1827](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1827/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit ce73f1cbc75412973e558fdffbc86657968f35ab (Job results: [1828](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1828/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit ce73f1cbc75412973e558fdffbc86657968f35ab (Job results: [1829](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1829/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 4cf4f807c0ee7056f342baa123be3e17393711b0 (Job results: [1830](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1830/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 4cf4f807c0ee7056f342baa123be3e17393711b0 (Job results: [1831](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1831/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:no_good: The commit e7c7b404cdccba0d8412673114f7651fef305e58 (Job results: [1752](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1752/)) failed the code audit or documentation test.

File: osquery/core/tests/posix/permissions_tests.cpp [hunk 2]
```diff
@@ -91,13 +93,13 @@ TEST_F(PermissionsTests, test_path_drop) {
...
 
     // Make sure the dropper worked!
     EXPECT_EQ(geteuid(), nobody->pw_uid);
   }
-
```
Comment (theopolis) [inline osquery/core/tests/posix/permissions_tests.cpp]: Why the collateral newline removal?

**Comment by osqueryer** [comment]
:-1: The commit f5b9fb521f28cac06f789ad596935c4aa07f2cf6 (Job results: [1835](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1835/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit f5b9fb521f28cac06f789ad596935c4aa07f2cf6 (Job results: [1836](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1836/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 7178ebf3366f4a39275ba8bad9ea443a04ebfc0e (Job results: [1837](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1837/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 7178ebf3366f4a39275ba8bad9ea443a04ebfc0e (Job results: [1838](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1838/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit ebaddfd475709b2ef43bac6dca4725356b73e9dc (Job results: [1839](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1839/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit ebaddfd475709b2ef43bac6dca4725356b73e9dc (Job results: [1840](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1840/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 10ec56a78d32c4fcfbbca57b5a5efa283ff12660 (Job results: [1842](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1842/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 10ec56a78d32c4fcfbbca57b5a5efa283ff12660 (Job results: [1843](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1843/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 04d998ffdeea86c744907d34c5cf4d08b37e341b (Job results: [1844](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1844/)) failed one or more tests (**Windows**).

File: osquery/core/system.cpp [hunk 3]
```diff
@@ -7,7 +7,6 @@
  *  of patent rights can be found in the PATENTS file in the same directory.
  *
  */
-
```
Comment (theopolis) [inline osquery/core/system.cpp]: 😿

File: osquery/core/system.cpp [hunk 4]
```diff
@@ -439,7 +438,6 @@ bool DropPrivileges::dropTo(uid_t uid, gid_t gid) {
     (void)setegid(getgid());
     return false;
   }
-
```
Comment (theopolis) [inline osquery/core/system.cpp]: 😿

File: osquery/tests/test_util.h [hunk 5]
```diff
@@ -21,6 +25,18 @@
 #include <osquery/database.h>
 #include <osquery/events.h>
 #include <osquery/filesystem.h>
+#ifndef WIN32
+#define AS_NOBODY(code)                                                        \
```
Comment (theopolis) [inline osquery/tests/test_util.h]: Please use a function or lambda.

File: osquery/core/tests/posix/permissions_tests.cpp [hunk 6]
```diff
@@ -9,7 +9,9 @@
  */
 
 #include <poll.h>
+#ifndef WIN32
```
Comment (theopolis) [inline osquery/core/tests/posix/permissions_tests.cpp]: Why? Isn't this in `./posix/`?

**Comment by osqueryer** [comment]
:no_good: The commit 3917012086cf6103576d0399298417b61595c08f (Job results: [1799](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1799/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 3917012086cf6103576d0399298417b61595c08f (Job results: [1893](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1893/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 3917012086cf6103576d0399298417b61595c08f (Job results: [1894](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1894/)) failed one or more tests (**Windows**).

**Comment by jedi22** [comment]
Ah my bad, I fixed these (except the lambda one) but I reverted the diff because using a function was breaking Windows.

**Pull request closed without merging**